pkg/operator/ceph/cluster/watcher.go

```diff
@@ -91,20 +91,18 @@ func (c *clientCluster) onK8sNode(object runtime.Object) bool {
 		// If so we don't need to reconcile, this is done to avoid double reconcile on operator restart
 		// Assume the admin key since we are watching for node status to create OSDs
 		clusterInfo := cephclient.AdminClusterInfo(cluster.Namespace)
-		osds, err := cephclient.GetOSDOnHost(c.context, clusterInfo, nodeName)
+		crushName := cephclient.NormalizeCrushName(nodeName)
+		osds, err := cephclient.GetOSDOnHost(c.context, clusterInfo, crushName)
 		if err != nil {
 			// If it fails, this might be due to the the operator just starting and catching an add event for that node
-			logger.Debugf("failed to get osds on node %q", nodeName)
-			return false
+			logger.Debugf("failed to get osds on node %q, assume reconcile is necessary", crushName)
+			return true
 		}
 
 		// If they are OSDs in the CRUSH map and if the host exists in the CRUSH map, don't reconcile
 		if osds != "" {
 			// This is Debug level because the node receives frequent updates and this will pollute the logs
 			logger.Debugf("node watcher: node %q is already an OSD node with %q", nodeName, osds)
-		} else {
-			logger.Infof("node watcher: adding node %q to cluster %q", node.Labels[v1.LabelHostname], cluster.Namespace)
```

Comment:
In your testing, does `GetOSDOnHost()` always fail and trigger `return true` on line 99 in case there are no OSDs? So there is no need for this case for `return true` anymore?

Reply:
Yes `GetOSDOnHost()` failed and was triggering `return true` on line 99. Okay I will update it back to return false.

Reply:
Yes @travisn there is the need for `return true`, as it will call reconcile to happen otherwise it will loop again over the same process and never return true from the `onK8sNode` function and so never reconcile. By `return true` it is resulting in reconcile and `osds-prepare` get created for added nodes.